mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt

```diff
@@ -368,7 +368,8 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
 
     override fun onPrepareOptionsMenu(menu: Menu): Boolean {
         CrashReportingHelper.d(TAG, "onPrepareOptionsMenu()")
-        menu.findItem(R.id.mainmenu_voice_recognition).isVisible = connection != null
+        val isSpeechRecognizerAvailable = SpeechRecognizer.isRecognitionAvailable(this)
+        menu.findItem(R.id.mainmenu_voice_recognition).isVisible = connection != null && isSpeechRecognizerAvailable
         menu.findItem(R.id.mainmenu_crash).isVisible = BuildConfig.DEBUG
```

Comment:
I think omitting the intermediate variable makes more sense, as that way the check can be skipped if `connection` is `null`.

Reply:
Done